cloud/vsphere/machineactuator.go

```diff
@@ -433,36 +433,47 @@ func runTerraformCmd(stdout bool, workingDir string, arg ...string) (bytes.Buffe
 	return out, nil
 }
 
-func (vc *VsphereClient) Delete(_ *clusterv1.Cluster, machine *clusterv1.Machine) error {
+func (vc *VsphereClient) Delete(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
 	// Check if the instance exists, return if it doesn't
 	instance, err := vc.instanceIfExists(machine)
 	if err != nil {
 		return err
 	}
-
 	if instance == nil {
 		return nil
 	}
 
-	homedir, err := getHomeDir()
+	clusterConfig, err := vc.clusterproviderconfig(cluster.Spec.ProviderConfig)
 	if err != nil {
 		return err
 	}
-	machinePath := path.Join(homedir, fmt.Sprintf(".terraform.d/kluster/machines/%s", machine.ObjectMeta.Name))
+
+	machinePath, err := vc.prepareStageMachineDir(machine)
+
+	// The machine HCL reads the startup script, so we need to have something there for terraform
+	// to not fail.
+	if err := saveFile("", path.Join("/tmp", "machine-startup.sh"), 0644); err != nil {
```

Comment:
It would be better to use a library driven temporary location so on systems where tmp is not a /tmp this code will not error, i.e. ioutil.TempFile(...):

https://golang.org/pkg/io/ioutil/#TempFile